main.go

```diff
@@ -144,8 +146,18 @@ func main() {
 			for k, v := range headers {
 				w.Header().Set(k, v)
 			}
-			w.WriteHeader(statusCode)
-			w.Write(body)
+			if statusCode == 200 {
+				// handle if-match, if-none-match request headers based on response etag
+				http.ServeContent(
+					w, r,
+					"",                // name used to infer content-type, but we've already set that
+					time.UnixMilli(0), // ignore setting last-modified time and handling if-modified-since headers
+					bytes.NewReader(body),
+				)
+			} else {
+				w.WriteHeader(statusCode)
+				w.Write(body)
+			}
```

Comment:
We might also need to add this into the caddy module... I'm not seeing a way to make caddy automatically add the if-match/if-none-match handling?